src/app/shared/change-password-dialog/change-password-dialog.component.html

```diff
@@ -0,0 +1,47 @@
+<h1 mat-dialog-title>Change Password</h1>
+
+<div mat-dialog-content>
+
+  <form [formGroup]="changePasswordForm">
+
+    <div fxLayout="row wrap" fxLayout.lt-md="column" fxLayoutGap="2%">
+
+      <mat-form-field fxFlex="96%">
+        <mat-label>Password</mat-label>
+        <input matInput formControlName="password" type="password">
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('required')">
+          Password <strong>is required</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('minlength')">
+          Password should be at least <strong>8 characters long</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('maxlength')">
+          Password should not be more than <strong>50 characters long</strong>
+        </mat-error>
+        <mat-error *ngIf="changePasswordForm.controls.password.hasError('pattern')">
+          Pattern does not match
+        </mat-error>
+      </mat-form-field>
+
+    </div>
+
+    <div fxLayout="row wrap" fxLayout.lt-md="column" fxLayoutGap="2%">
+
+      <mat-form-field fxFlex="96%">
+        <mat-label>Repeat Password</mat-label>
+        <input matInput formControlName="repeatPassword" type="password">
+        <mat-error *ngIf="changePasswordForm.controls.repeatPassword.hasError('notequal')">
+          Password <strong>does not match</strong>
+        </mat-error>
+      </mat-form-field>
+
+    </div>
+
+  </form>
+
+</div>
+
+<mat-dialog-actions align="end">
+  <button mat-raised-button mat-dialog-close>Cancel</button>
+  <button mat-raised-button color="warn" [disabled]="!changePasswordForm.valid" [mat-dialog-close]="{password:changePasswordForm.get('password').value,repeatPassword:changePasswordForm.get('repeatPassword').value}">Confirm</button>
```

Comment:
get mat dialog close attribute as a return value of a function

Reply:
It's been used